### PR TITLE
Support for 1024b RSA SHA-512 keys used in DNSSEC

### PR DESCRIPTION
**Pull request by vavrusa**
See https://github.com/briansmith/ring/issues/870. This adds `RSA_PKCS1_1024_8192_SHA512_FOR_LEGACY_USE_ONLY` for DNSSEC verification as it's still used by several TLDs.

### CODE REVIEW

**Comment by vavrusa** [comment]
@briansmith I've tried to generate test vectors with convert_nist_rsa_test_vectors.py but couldn't find documentation on how to use it (and what's the input), any pointers would be appreciated.

File: src/rsa/convert_nist_rsa_test_vectors.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def print_sign_test(case, n, e, d, padding_alg):
         # PSS is randomised, can't recompute this way
         pass
     else:
-        print "Invalid padding algorithm"
```
Comment (vavrusa) [inline src/rsa/convert_nist_rsa_test_vectors.py]: I've added this in a separate commit to make the script compatible with Python 3 as I can't install Python 2.7 from homebrew anymore, but I can remove this if you don't want this.

File: src/rsa/convert_nist_rsa_test_vectors.py [hunk 2]
```diff
@@ -207,8 +207,8 @@ def main(fn, test_type, padding_alg):
 
             print_sign_test(case, n, e, d, padding_alg)
         else:
-            legacy = case['SHAAlg'] in ["SHA1", "SHA256"]
-            if (n.bit_length() < 2048 and not legacy) or n.bit_length() < 1024:
```
Comment (vavrusa) [inline src/rsa/convert_nist_rsa_test_vectors.py]: I've added this change to allow odd length keys padded to at least 1024/2048b.

**Comment by vavrusa** [comment]
Rebased

**Comment by Darkspirit** [comment]
https://github.com/bluejekyll/trust-dns/issues/907 depends on this, too.

**Comment by briansmith** [comment]
@vavrusa Please ammend each commit message to contain the contributor agreement statement at https://github.com/briansmith/ring/#contributing. Then I will take a look at this. (Sorry for the inconvenience.)

**Comment by vavrusa** [comment]
@briansmith no problem, amended

**Comment by briansmith** [comment]
Thanks! Merged!

**Pull request merged**